adapters/onebot/src/bot/message.ts

```diff
@@ -43,17 +44,28 @@ export class OneBotMessenger extends Messenger<BaseBot> {
     }
 
     // flush
-    if (!this.children.length) return
     const { type, author } = this.stack[0]
+    if (!this.children.length && !author.messageId) return
     if (type === 'forward') {
-      this.stack[1].children.push({
-        type: 'node',
-        data: {
-          name: author.nickname || author.username || this.bot.nickname || this.bot.username,
-          uin: author.userId || this.bot.userId,
-          content: this.children as any,
-        },
-      })
+      if (author.messageId) {
+        this.stack[1].children.push({
+          type: 'node',
+          data: {
+            id: author.messageId
+          },
+        })
+      } else {
+        this.stack[1].children.push({
+          type: 'node',
+          data: {
+            name: author.nickname || author.username || this.bot.nickname || this.bot.username,
+            uin: author.userId || author['user-id'] || this.bot.userId,
+            content: this.children as any,
+            time: `${(+author.time || Date.now()) / 1000}`
```

Comment:
```suggestion
            uin: author.userId || this.bot.userId,
            content: this.children as any,
            time: `${Math.floor((+author.time || Date.now()) / 1000)}`,
```

Not sure if go-cqhttp supports this.